cli/templatepush.go

```diff
@@ -164,24 +166,33 @@ func (r *RootCmd) templatePush() *clibase.Cmd {
 				return err
 			}
 
-			template, err := client.TemplateByName(inv.Context(), organization.ID, name)
+			resp, err := uploadFlags.upload(inv, client)
 			if err != nil {
 				return err
 			}
 
-			err = uploadFlags.checkForLockfile(inv)
-			if err != nil {
-				return xerrors.Errorf("check for lockfile: %w", err)
-			}
-
-			resp, err := uploadFlags.upload(inv, client)
+			tags, err := ParseProvisionerTags(provisionerTags)
 			if err != nil {
 				return err
 			}
 
-			tags, err := ParseProvisionerTags(provisionerTags)
+			template, err := client.TemplateByName(inv.Context(), organization.ID, name)
 			if err != nil {
-				return err
+				if !create {
+					_, _ = fmt.Fprintf(inv.Stdout, "Create a new template with `coder templates create %s`.\n", name)
+					_, _ = fmt.Fprintf(inv.Stdout, "Or use `coder templates push %s --create`.\n", name)
+					return xerrors.Errorf("template %q not found: %w", name, err)
+				}
+				_, _ = fmt.Fprintf(inv.Stdout, "Creating a new template: %s\n", name)
+				defaultTTL := 24 * time.Hour
+				failureTTL := 0 * time.Hour
+				inactivityTTL := 0 * time.Hour
+				disableEveryone := false
+				createReq := codersdk.CreateTemplateRequest{Name: name, DefaultTTLMillis: ptr.Ref(defaultTTL.Milliseconds()), FailureTTLMillis: ptr.Ref(failureTTL.Milliseconds()), InactivityTTLMillis: ptr.Ref(inactivityTTL.Milliseconds()), DisableEveryoneGroupAccess: disableEveryone}
+				template, err = client.CreateTemplate(inv.Context(), organization.ID, createReq)
```

Comment:
This line specifically. The object `template` isn't being updated as expected. 
@johnstcn

Reply:
Missing CreateTemplateRequest.VersionID - creating a template requires a template version to be present.